druid-shell/src/platform/x11/window.rs

```diff
@@ -171,14 +160,10 @@ impl WindowBuilder {
             &cairo_visual_type,
             self.size.width as i32,
             self.size.height as i32,
-        );
-        match cairo_surface {
-            Ok(cairo_surface) => Ok(RefCell::new(cairo::Context::new(&cairo_surface))),
-            Err(err) => Err(Error::Generic(format!(
-                "Failed to create cairo surface: {}",
-                err
-            ))),
-        }
+        )
+        .map_err(|status| anyhow!("Failed to create cairo surface: {}", status))?;
```

Comment:
What's the benefit of `map_err` over `context`?

Reply:
`context` is only available on a `Result<T, E> where E: std::error::Error`, and several of the cairo "errors" don't implement `Error`